### DISCUSSION TITLE
Fred's Imagemagick Scripts - on Windows

### DISCUSSION THREAD

**Discussion by myicq** [ImageMagick/ImageMagick#6564] [General] [answered]
I am a bit fan of [Fred's imagemagick scripts](http://www.fmwconcepts.com/imagemagick/index.php). Recently I have discovered, thanks to a helpful pointer on the main page, and snigbo's comments, that the scripts can be run directly from windows 10 terminal using bash.

But! The help text for some reason prints every line double.

```none
embroidery:
USAGE: embroidery [-n numcolors ] [-p pattern] [-t thickness] [-g graylimit]
USAGE: embroidery [-n numcolors ] [-p pattern] [-t thickness] [-g graylimit]
[-f fuzzval] [-b bgcolor] [-P process] [-a angle] [-r range] [-i intensity] [-e extent]
[-f fuzzval] [-b bgcolor] [-P process] [-a angle] [-r range] [-i intensity] [-e extent]
```

**solution** to this is simple: edit the scripts and change the functions `usage1()` and `usage2()`, removing the text ` 4,$p`

usage1() should be
   `sed >&2 -e '1,/^####/d;  /^###/g;  /^#/!q;  s/^#//;  s/^ //; ' "$PROGDIR/$PROGNAME"`

Now the help prints correctly.
```none
embroidery:
USAGE: embroidery [-n numcolors ] [-p pattern] [-t thickness] [-g graylimit]
[-f fuzzval] [-b bgcolor] [-P process] [-a angle] [-r range] [-i intensity] [-e extent]
```

Just wanted to mention if someone else can use the tip.

**Comment by fmw42** [comment, marked as answer]
Yes, some unix systems have this issue and others do not.  It works fine on Mac OSX.  You will have to fix it as you suggest for each script you want to use.  At this time, I do not have the time to try to fix each of my 300+ scripts.

But thanks for post that.  However, I have solutions posted to my imagemagick script home page if you had looked. It was already there.

**Reply by myicq** [reply]
Thank you the response. You are right, it was there all the time. My bad! Missed it because all the page is proportional font, not monospace. Should I delete this post, since the information was there.

**Comment by fmw42** [comment]
No, don't delete this post.  It is good to have it here also.
